### PR TITLE
#16 add shoulda and some tests

### PR DESCRIPTION
**Pull request by Mywe4ka**

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -6,6 +6,9 @@
       @user = FactoryGirl.create(:user)
   end
 
+  it { should use_before_filter(:find_user) }
+  it { should_not use_before_filter(:set_user) }
```
Comment (VeronikaSalo) [inline spec/controllers/users_controller_spec.rb]: @Mywe4ka please explain this row. Do we have set_user method?

File: spec/models/song_spec.rb [hunk 2]
```diff
@@ -2,8 +2,19 @@
 
 describe Song do
 
-it 'has valid factory' do
-    expect(FactoryGirl.create(:song)).to be_valid
+  before do
```
Comment (VeronikaSalo) [inline spec/models/song_spec.rb]: @Mywe4ka let's try to convert it to `let(:song) {...}`

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@
 require 'spec_helper'
 require File.expand_path('../../config/environment', __FILE__)
 require 'rspec/rails'
+require 'shoulda/matchers'
```
Comment (VeronikaSalo) [inline spec/rails_helper.rb:6]: @Mywe4ka Doesn't it work without it? only by adding to spec_helper.rb ?

File: spec/controllers/users_controller_spec.rb (hunk 1 above)
Comment (Mywe4ka) [inline spec/controllers/users_controller_spec.rb]: No, we don't.
It is just a little check if  'use_before_filter' works properly.

File: spec/rails_helper.rb (hunk 3 above)
Comment (Mywe4ka) [inline spec/rails_helper.rb:6]: It doesn't work without it.
Shoulda needs knowing about test framework we're using.
That's why we require the gem following rspec-rails in rails_helper.rb || spec_helper.rb
Probably, the other way of configuring is https://github.com/thoughtbot/shoulda-matchers#configuration

**Pull request merged**